test/constructor-test.js

```diff
@@ -108,6 +108,22 @@ describe('BN.js/Constructor', function () {
         res;
       }, /Invalid character/);
     });
+
+    it('should not accept non-hex characters', function () {
+      [
+        '0000000z',
+        '000000gg',
+        '0000gg00',
+        'ffffggff',
+        'ffffggff',
+        'hexadecimal'
```

Comment:
Let's add `.`, `-`, `/`. They all come before `0` in ASCII.

Reply:
@indutny added, and it passes!